pkg/controller/vitessbackupschedule/vitessbackupschedule_controller.go

```diff
@@ -466,76 +549,83 @@ func (r *ReconcileVitessBackupsSchedule) createJob(ctx context.Context, vbsc *pl
 		Spec: kbatch.JobSpec{
 			Template: corev1.PodTemplateSpec{
 				ObjectMeta: meta,
-				Spec:       pod,
+				Spec:       pod.Spec,
 			},
 		},
 	}
 
-	if err := ctrl.SetControllerReference(vbsc, job, r.scheme); err != nil {
+	if err := ctrl.SetControllerReference(&vbsc, job, r.scheme); err != nil {
 		return nil, err
 	}
 
-	return job, nil
-}
-
-func (r *ReconcileVitessBackupsSchedule) createJobPod(ctx context.Context, vbsc *planetscalev2.VitessBackupSchedule, name string) (pod corev1.PodSpec, err error) {
-	getVtctldServiceName := func(cluster string) (string, error) {
-		vtctldServiceName, vtctldServicePort, err := r.getVtctldServiceName(ctx, vbsc, cluster)
+	// Create the corresponding PVC for the new vtbackup pod
+	pvc := &corev1.PersistentVolumeClaim{}
+	key := client.ObjectKey{
+		Namespace: job.Namespace,
+		Name:      name,
+	}
+	err = r.client.Get(ctx, key, pvc)
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			return nil, err
+		}
+		newPVC := vttablet.NewPVC(key, vtbackupSpec.TabletSpec)
+		if err := ctrl.SetControllerReference(&vbsc, newPVC, r.scheme); err != nil {
+			return nil, err
+		}
+		err = r.client.Create(ctx, newPVC)
 		if err != nil {
-			return "", err
+			return nil, err
 		}
-		return fmt.Sprintf("--server=%s:%d", vtctldServiceName, vtctldServicePort), nil
 	}
 
-	// It is fine to not have any default in the event there is no strategy as the CRD validation
-	// ensures that there will be at least one item in this list. The YAML cannot be applied with
-	// empty list of strategies.
-	var cmd strings.Builder
+	return job, nil
+}
 
-	addNewCmd := func(i int) {
-		if i > 0 {
-			cmd.WriteString(" && ")
-		}
+func (r *ReconcileVitessBackupsSchedule) createJobPod(
+	ctx context.Context,
+	vbsc planetscalev2.VitessBackupSchedule,
+	strategy planetscalev2.VitessBackupScheduleStrategy,
+	name string,
+	vkr planetscalev2.VitessKeyRange,
+	labels map[string]string,
+) (pod *corev1.Pod, spec *vttablet.BackupSpec, err error) {
+	vts, err := r.getShardFromKeyspace(ctx, vbsc.Namespace, vbsc.Spec.Cluster, strategy.Keyspace, strategy.Shard)
+	if err != nil {
+		return nil, nil, err
 	}
 
-	for i, strategy := range vbsc.Spec.Strategy {
-		vtctldclientServerArg, err := getVtctldServiceName(vbsc.Spec.Cluster)
-		if err != nil {
-			return corev1.PodSpec{}, err
-		}
+	_, completeBackups, err := vitessbackup.GetBackups(ctx, vbsc.Namespace, vbsc.Spec.Cluster, strategy.Keyspace, vkr.SafeName(),
```

Comment:
Nit, but I think the var name should be `completedBackups`, no?

Reply:
fixed via 072d109